src/foundry/client/pixi/sources/vision-source.d.ts

```diff
@@ -1,123 +1,155 @@
+    get fov(): PointSourcePolygon | PIXI.Polygon;
 
     /**
-     * The ratio of dim:bright as part of the source radius
-     * @defaultValue `undefined`
+     * If this vision source background is rendered into the lighting container.
      */
-    ratio: number | undefined;
+    get preferred(): boolean;
+
+    override get isAnimated(): boolean;
+
+    protected override _initialize(data: Partial<VisionSourceData>): void;
+
+    protected override _configure(changes: Partial<VisionSourceData>): void;
+
+    protected override _configureLayer(layer: RenderedPointSource.RenderedPointSourceLayer, layerId: string): void;
+
+    /** Responsible for assigning the Vision Mode and handling exceptions based on vision special status. */
+    protected _initializeVisionMode(): void;
+
+    /** {@inheritdoc} */
+    override _getPolygonConfiguration(): PointSourcePolygonConfig;
+
+    /** Create a restricted FOV polygon by limiting the radius of the unrestricted LOS polygon. */
+    protected _createRestrictedPolygon(): PointSourcePolygon;
+
+    protected override _configureShaders(): {
+      background: AdaptiveLightingShader;
+      coloration: AdaptiveLightingShader;
+      illumination: AdaptiveLightingShader;
+    };
 
     /**
-     * The rendered field-of-vision texture for the source for use within shaders.
-     * @defaultValue `undefined`
+     * Update shader uniforms by providing data from this VisionSource
      */
-    fovTexture: PIXI.RenderTexture | undefined;
+    protected _updateColorationUniforms(): void;
 
     /**
-     * Track which uniforms need to be reset
-     * @defaultValue `{ illumination: true }`
-     * @internal
+     * Update shader uniforms by providing data from this VisionSource
      */
-    protected _resetUniforms: { illumination: boolean };
+    protected _updateIlluminationUniforms(): void;
 
     /**
-     * To track if a source is temporarily shutdown to avoid glitches
-     * @defaultValue `{ illumination: false }`
-     * @internal
+     * Update shader uniforms by providing data from this VisionSource
      */
-    protected _shutdown: { illumination: boolean };
+    protected _updateBackgroundUniforms(): void;
 
     /**
-     * Initialize the source with provided object data.
-     * @param data - Initial data provided to the point source
-     * @returns A reference to the initialized source
+     * Update shader uniforms shared by all shader types
+     * @param shader - The shader being updated
+     * @internal
      */
-    initialize(data?: Partial<VisionSourceData>): this;
-
-    fov?: PIXI.Circle;
+    _updateCommonUniforms(shader: AdaptiveVisionShader): void;
 
     /**
-     * Initialize the blend mode and vertical sorting of this source relative to others in the container.
-     * @internal
+     * TODO: Replace any with the type inherited from _configureLayer
+     * const vmUniforms = this.visionMode.vision[layerId].uniforms;
+     * layer.vmUniforms = Object.entries(vmUniforms);
```

Comment:
The formatting is weird since it's not in a code block. Also what's holding up implementing this? I can help write the type if you need but `Object.entries` can (basically) be typed as `Obj[keyof Obj]` so `typeof vmUniforms[keyof typeof vmUniforms]`.